response.go

```diff
@@ -2,11 +2,36 @@ package opensearchtools
 
 import "net/http"
 
+// RawOpenSearchResponse is a generic return type for an OpenSearch query which has meta fields common
+// to all response types as well as a generic Response field abstract across all response types.
+type RawOpenSearchResponse[T any] struct {
```

Comment:
I don't really like the naming of `RawOpenSearchResponse` and `OpenSearchResponse` because "raw" is a bit misleading (makes me think of the raw response we get back from OS, but that's not really what this is).

My other thought was `OpenSearchResponse` and `OpenSearchResponseWithValidation` which is nice because it is accurate, is getting a bit long. LMKWYT.

Reply:
I mentioned it in my comment. But I don't think we need this struct at all. I'm not sure I see the value this adds by separating out the validation results and having two "Response" types.

I'd maybe like it better if `OpenSearchResponse` extend `RawOpenSearchResponse` instead of having it as an attribute. But it still seems a little clunky to me.

Reply:
I agree with @cs-marshall.  Don't particularly like the extra type.  Why not just keep `OpenSearchResponse[T any]` as we had it? 😕

A response with no validation result will just have the `ValidationResults` field set to `nil`.  No need for a separate type, IMO.